discrete_math/polynomial/polynomial.cxx

```diff
@@ -0,0 +1,289 @@
+module;
+#include <algorithm>
+#include <cmath>
+#include <cstdio>
+#include <format>
+#include <iostream>
+#include <numeric>
+#include <set>
+#include <stdexcept>
+#include <utility>
+#include <vector>
+
+export module polynomial;
+
+
+export namespace polynomial {
+    template <typename T>
+    struct polynomial {
+      public:
+        std::vector<T> coefficients{};
+        std::size_t degree{};
```

Comment:
I changed so that polynomial doesn't hold degree value but we can get it by polynomial.degree(), similar to matrix.number_of_rows().
Made construct that construct polynomial of given degree with zero coefficients

Reply:
still you do not satisfy e.g. 1. point that is you cannot write polynomial{1,2,3} to create polynomial with coeffs 1,2,3....

Reply:
i made so that you can create polynomial using degree polynomial{2} with all coefficients 0 so it prints 0, and using coefficients polynomial{{1, 2, 3}} where it would print x^2 + 2x + 3

Reply:
yeah but it really is natural to define polynomial via polynomial{1,2,3} so there should be such a possibility..

Reply:
but it is implemented in line 33. std::print("{}\n", polynomial{{1, 2, 3}}); will print: x^2 + 2x + 3